sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/WebPubSubConnectionContext.cs

```diff
@@ -78,9 +89,41 @@ public class WebPubSubConnectionContext
         /// <param name="userId">User Id.</param>
         /// <param name="signature">Signature of the connection.</param>
         /// <param name="origin">Origin of the event.</param>
-        /// <param name="states">Connection states. Value in <see cref="JsonElement"/> type. Use the <see cref="TryGetState{T}(string, out T)"/> method to tey getting the vaule deserialized as a specific type.</param>
+        /// <param name="states">Connection states. Value in <see cref="BinaryData"/> type.</param>
         /// <param name="headers">Connection request headers.</param>
-        public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName, string hub, string connectionId, string userId = null, string signature = null, string origin = null, IReadOnlyDictionary<string, object> states = null, IReadOnlyDictionary<string, string[]> headers = null)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public WebPubSubConnectionContext(WebPubSubEventType eventType, string eventName, string hub, string connectionId, string userId, string signature, string origin, IReadOnlyDictionary<string, object> states, IReadOnlyDictionary<string, string[]> headers)
+            : this(
+                eventType,
+                eventName,
+                hub,
+                connectionId,
+                userId,
+                signature,
+                origin,
+                states?.ToDictionary(
+                    p => p.Key,
+                    p => p.Value is BinaryData data ?
+                        data :
+                        // TODO: Is this safe?  Will all callers of of this .ctor always pass BinaryData? Can we have version mismatches between Common and Extensions that break this?
+                        throw new ArgumentException($"{nameof(states)} values must be of type {nameof(BinaryData)}.", nameof(states))),
+                headers)
+        {
+        }
```

Comment:
We should also chat about this TODO.  At this point I'm thinking it's safe, but it'd be good to verify end to end.

Reply:
The only breaking case is work with 1.0.0 Extension but 1.1.0+ Common. Theoretically could be though should be rare. Or for safety, we check type to convert to BinaryData if original type is not BinaryData?

Reply:
That's a really good idea.  I was already doing that in the older `SetState(string key, object value)` overload.  It's probably the best reasonable compromise and I agree it's very unlikely for this to break anyone.